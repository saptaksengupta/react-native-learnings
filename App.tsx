import React from 'react';
import { View, Text } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import Navigation from './features/navigation';

import useColorScheme from './hooks/useColorScheme';

export default function App() {
  const colorScheme = useColorScheme();
  return (
    <SafeAreaProvider>
      <Navigation colorScheme={colorScheme}/>
    </SafeAreaProvider>
  )
}

