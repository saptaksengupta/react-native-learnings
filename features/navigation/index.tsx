import {
  NavigationContainer,
  DefaultTheme,
  DarkTheme,
} from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import * as React from "react";
import { ColorSchemeName } from "react-native";

import { RootStackParamList } from "../../shared/types/core.type";
import LinkingConfiguration from "./LinkingConfiguration";

export default function Navigation({
  colorScheme,
}: {
  colorScheme: ColorSchemeName;
}) {
  return (
    <NavigationContainer
      linking={LinkingConfiguration}
      theme={colorScheme === "dark" ? DarkTheme : DefaultTheme}
    >
      <RootNavigator />
    </NavigationContainer>
  );
}

const Stack = createStackNavigator<RootStackParamList>();

function RootNavigator() {
  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
    </Stack.Navigator>
  );
}
