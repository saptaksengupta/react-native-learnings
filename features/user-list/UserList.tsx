import * as React from 'react';
import { Text } from "react-native";

export const UserList = (): JSX.Element => {
    return (
        <Text>Uset List feature and test added</Text>
    )    
}