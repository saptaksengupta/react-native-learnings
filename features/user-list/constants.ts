export const actions = {
    FETCH_USER: {
        REQUEST: 'FETCH_USER.REQUEST', 
        SUCCESS: 'FETCH_USER.SUCCESS', 
        ERROR: 'FETCH_USER.ERROR', 
    }
}

export interface IUserListStateType {
    /**
     * Flag to determine fetch user is in progress or not.
     */
    fetchUserLoading: boolean;

    /**
     * User object
     */
    user: Object | null;

    /**
     * Fetch user error 
     */
    fetchUserError: string | null

}