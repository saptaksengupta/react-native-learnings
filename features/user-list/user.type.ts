export interface IUserType {
    id: string;
    nickName: string;
    avatarSm: string;
    phoneNumber: string;
    status: number;
}