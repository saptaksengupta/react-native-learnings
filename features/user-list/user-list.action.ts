
import { actions } from './constants';

/**
 * 
 * Action to request users.
 *  
 */
export const fetchUsers = () => ({
    type: actions.FETCH_USER.REQUEST,
    payload: null
})

/**
 * 
 * @param users 
 * Action to notify user fetch success.
 */
export const fetchUsersSuccess = (users) => ({
    type: actions.FETCH_USER.SUCCESS,
    payload: users
})

/**
 * 
 * @param error 
 * Action to notidy fetch user failure
 */

export const fetchUsersError = (error) => ({
    type: actions.FETCH_USER.ERROR,
    payload: error
});


export type UserListActions = ReturnType<
    typeof fetchUsersSuccess |
    typeof fetchUsersError |
    typeof fetchUsers>
