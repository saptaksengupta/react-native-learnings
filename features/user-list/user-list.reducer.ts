import { IUserListStateType, actions } from './constants';

import { UserListActions } from './user-list.action';

const initialState: IUserListStateType = {
    user: null,
    fetchUserError: null,
    fetchUserLoading: false
}

export const userListReducer = (state = initialState, action: UserListActions): IUserListStateType => {
    switch (action.type) {
        case actions.FETCH_USER.REQUEST:
            return {
                fetchUserLoading: true,
                user: null,
                fetchUserError: null
            }
        case actions.FETCH_USER.SUCCESS:
            return {
                fetchUserLoading: false,
                user: action.payload,
                fetchUserError: null
            }
        case actions.FETCH_USER.ERROR: 
            return {
                fetchUserLoading: false,
                user: null,
                fetchUserError: action.payload
            }
        default:
            return state;
    }
}