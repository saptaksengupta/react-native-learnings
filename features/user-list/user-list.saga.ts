import { actions } from './constants';
import { all, takeEvery } from "redux-saga/effects";

function *watchFetchUsers() {
    yield takeEvery(actions.FETCH_USER.REQUEST, fetchUsers);
} 

function *fetchUsers() {
    yield console.log('Hey its inside user saga');
}


export default function *fetchUserSagas() {
    yield all([
        watchFetchUsers()
    ])
}