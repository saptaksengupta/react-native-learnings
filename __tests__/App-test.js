/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { UserList } from '../features/user-list';

it('renders user list correctly', () => {
  const tree = renderer.create(<UserList />);
  expect(tree).toMatchSnapshot();
});
