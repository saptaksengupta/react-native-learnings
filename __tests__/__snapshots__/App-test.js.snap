// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user list correctly 1`] = `
<Text>
  Uset List feature and test added
</Text>
`;
